powersimdata/data_access/data_access.py

```diff
@@ -279,101 +181,59 @@ def get_profile_version(self, grid_model, kind):
             if should_attempt:
                 try:
-                    self._setup_server_connection()
-                    return self._ssh
+                    self._fs = get_ssh_fs(self.root)
+                    return self._fs
                 except:  # noqa
                     SSHDataAccess._last_attempt = time.time()
             msg = f"Could not connect to server, will try again after {self._retry_after} seconds"
             raise IOError(msg)
 
-        return self._ssh
-
-    def _setup_server_connection(self):
-        """This function setup the connection to the server."""
-        client = paramiko.SSHClient()
-        client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        try:
-            client.load_system_host_keys()
-        except IOError:
-            print("Could not find ssh host keys.")
-            ssh_known_hosts = input("Provide ssh known_hosts key file =")
-            while True:
-                try:
-                    client.load_system_host_keys(str(ssh_known_hosts))
-                    break
-                except IOError:
-                    print("Cannot read file, try again")
-                    ssh_known_hosts = input("Provide ssh known_hosts key file =")
-
-        server_user = server_setup.get_server_user()
-        client.connect(
-            server_setup.SERVER_ADDRESS,
-            username=server_user,
-            port=server_setup.SERVER_SSH_PORT,
-            timeout=10,
-        )
-
-        self._ssh = client
+        return self._fs
 
     def copy_from(self, file_name, from_dir=None):
         """Copy a file from data store to userspace.
 
         :param str file_name: file name to copy.
         :param str from_dir: data store directory to copy file from.
         """
-        self._check_filename(file_name)
         from_dir = "" if from_dir is None else from_dir
-        to_dir = os.path.join(self.local_root, from_dir)
-        os.makedirs(to_dir, exist_ok=True)
-
-        from_path = self.join(self.root, from_dir, file_name)
-        to_path = os.path.join(to_dir, file_name)
+        from_path = self.join(from_dir, file_name)
         self._check_file_exists(from_path, should_exist=True)
 
-        with self.ssh.open_sftp() as sftp:
-            print(f"Transferring {file_name} from server")
-            cbk, bar = progress_bar(ascii=True, unit="b", unit_scale=True)
-            tmp_file, tmp_path = mkstemp()
-            sftp.get(from_path, tmp_path, callback=cbk)
-            bar.close()
-            os.close(tmp_file)
-        # wait for file handle to be available
-        shutil.move(tmp_path, to_path)
+        print(f"Transferring {file_name} from server")
+        with TempFS() as tmp_fs:
+            self.local_fs.makedirs(from_dir, recreate=True)
+            tmp_fs.makedirs(from_dir, recreate=True)
+            fs2.copy.copy_file(self.fs, from_path, tmp_fs, from_path)
+            fs2.move.move_file(tmp_fs, from_path, self.local_fs, from_path)
```

Comment:
This is clean! I presume it is for the purpose if conflict protection, right?

Reply:
Yeah, this was probably my favorite part. And that is correct - the idea is to minimize the time we have the destination file open (e.g. ScenarioList.csv) to reduce collisions when running concurrently. Another benefit is if the download fails, e.g. for some large file, we don't end up with a partial copy.